### DISCUSSION TITLE
Allow gacela-env.php

### DISCUSSION THREAD

**Discussion by JesusValeraDev** [gacela-project/gacela#88] [Ideas]
Currently, it's only possible to define in the root the `gacela.php` config file.
But it could be interesting if we want to override the `mappingInterfaces()` method and use another implementation in testing for example.

This about this idea if makes sense to implement it or not.

**Comment by Chemaclass** [comment]
As I understood, what you mean is about creating different `gacela.php` files (or at least allow it) for each environment. Such as:
- `gacela.php` <- the default gacela setup
- `gacela-test.php` <- for automated testing, for example

It wouldn't be hard at all. The place where we read the `gacela.php` file is `GacelaConfigFileFactory->createGacelaFileConfig()`. We could inject the APP_ENV inside that Factory and use it to check if we have an especific gacela file with that suffix or not. And then merge the values from the "gacela environment specific file" with the default gacela (to allow partial overrides only for the specific).

> But it could be interesting if we want to override the mappingInterfaces() method and use another implementation in testing for example.

I think this is a great example. We could use the mappingInterfaces to map "DBConnection" or "Logger" to different concrete classes depending on the environment. For automated testing we could map to fake concretions, while on development to the real class.

I can't think about other usages. For example, `gacela-stage.php` or `gacela-prod.php`, I don't see a benefit or a reason why they should be different as the one in my local machine where I develop the features... I see the benefits only for the automated tests. Even thought I am not 100% sure it will work, it might worth a try. Do you have some other ideas, or do you want to create an issue with details about this, @JesusValera?

**Comment by Chemaclass** [comment]
This was already implemented here https://github.com/gacela-project/gacela/pull/81 & https://github.com/gacela-project/gacela/pull/181
